asset/src/spaces_reader_api/schema.ts

```diff
@@ -1,15 +1,59 @@
-import { ConvictSchema, AnyObject, cloneDeep } from '@terascope/job-components';
-import { spacesSchema } from '../spaces_reader/schema';
+import {
+    ConvictSchema,
+    AnyObject,
+    getTypeOf,
+    cloneDeep,
+    isNumber,
+    ValidatedJobConfig
+} from '@terascope/job-components';
+import elasticAPI from '@terascope/elasticsearch-api';
+import { schema } from '../elasticsearch_reader_api/schema';
 import { ApiConfig } from '../elasticsearch_reader/interfaces';
+import { DEFAULT_API_NAME } from './interfaces';
+
+const clone = cloneDeep(schema);
+
+const apiSchema = {
+    endpoint: {
+        doc: 'The base API endpoint to read from: i.e. http://yourdomain.com/api/v1',
+        default: null,
+        format: 'required_String'
+    },
+    token: {
+        doc: 'API access token for making requests',
+        default: null,
+        format: 'required_String'
+    },
+    timeout: {
+        doc: 'Time in milliseconds to wait for a connection to timeout.',
+        default: 300000,
+        format(val: unknown): void {
+            if (!isNumber(val)) throw new Error(`Invalid parameter timeout, it must be of type number, was given ${getTypeOf(val)}`);
+            if (isNaN(val)) throw new Error('Invalid timeout parameter, must be a number');
+            if (val <= 0) throw new Error('Invalid timeout parameter, must be greater than zero');
+        }
```

Comment:
Using the built-in `duration` is more user friendly, it works with human readable formats or milliseconds

```suggestion
        default: '30 seconds',
        format: 'duration'
```